community/kernel/src/main/java/org/neo4j/kernel/impl/factory/GraphDatabaseFacade.java

```diff
@@ -650,11 +719,103 @@ private InternalTransaction beginTransactionInternal( KernelTransaction.Type typ
             }
         }
 
-        return getNodesByLabelAndPropertyWithoutIndex( propertyId, value, statement, labelId );
+        return getNodesByLabelAndPropertyWithoutIndex( statement, labelId, query );
+    }
+
+    private ResourceIterator<Node> nodesByLabelAndProperties( KernelTransaction transaction, int labelId, IndexQuery... queries )
+    {
+        Statement statement = transaction.acquireStatement();
+        Read read = transaction.dataRead();
+
+        if ( isInvalidQuery( labelId, queries ) )
+        {
+            statement.close();
+            return emptyResourceIterator();
+        }
+
+        int[] propertyIds = getSortedPropertyIds( queries );
+        int[] workingCopy = new int[propertyIds.length];
+
+        Iterator<CapableIndexReference> indexes = transaction.schemaRead().indexesGetForLabel( labelId );
```

Comment:
we have a method 'storeReadLayer.indexGetForSchema' that know how to find index by label and all property keys, maybe using that instead will be actually faster than looking for each matching index separately?

Reply:
Yes, the `storeaReadLayer.indexGetForSchema` is reachable by `SchemaRead.index`, which I now added as a first attempt at a fast path. However this method does not try to reorder properties because no-one else needs that, so we still need to retain the `indexesGetForLabel` to try to find out-of-order indexes.